### PR TITLE
chore: move function saving metrics event to gateway

### PR DESCRIPTION
**Pull request by masaaania**
- move function saving metrics event to the api-gateway
- add new metrics event according to HTTP status code
- add e2e test for metrics event saving

### CODE REVIEW

File: pkg/gateway/api/api_grpc.go [hunk 1]
```diff
@@ -769,12 +773,23 @@ func (s *grpcGatewayService) RegisterEvents(
 				}
 				continue
 			}
-			metricsMessages = append(metricsMessages, event)
+			m := &eventproto.MetricsEvent{}
+			if err := ptypes.UnmarshalAny(req.Events[i].Event, m); err != nil {
```
Comment (ono-max) [inline pkg/gateway/api/api_grpc.go:777]: Can we write this code like this?
```suggestion
			if err := ptypes.UnmarshalAny(event.Event, m); err != nil {
```

File: pkg/gateway/api/api_grpc.go [hunk 2]
```diff
@@ -794,6 +809,18 @@ func (s *grpcGatewayService) containsInvalidTimestampError(errs map[string]*gwpr
 	return false
 }
 
+func (s *grpcGatewayService) mergeMaps(
```
Comment (ono-max) [inline pkg/gateway/api/api_grpc.go:812]: nit
```suggestion
func (*grpcGatewayService) mergeMaps(
```

File: pkg/gateway/api/api_grpc.go (hunk 1 above)
Comment (masaaania) [reply pkg/gateway/api/api_grpc.go:777]: https://qiita.com/uchiko/items/1c611f0db618ce9dc0a9

File: pkg/gateway/api/metrics_event.go [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+func (s *grpcGatewayService) saveUnknownError(event *eventproto.MetricsEvent, env string) error {
+	errorType := ErrorTypeUnknown
+	ev := &eventproto.UnknownErrorMetricsEvent{}
+	if err := ptypes.UnmarshalAny(event.Event, ev); err != nil {
+		return err
+	}
+	if ev.ApiId == eventproto.ApiId_UNKNOWN_API {
+		return MetricsSaveErrUnknownApiId
+	}
+	var tag string
```
Comment (ono-max) [inline pkg/gateway/api/metrics_event.go:500]: Do we need to insert "none" if tag is empty?
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/eventpersisterdwh/persister/evaluation_event.go#L166

Comment (cre8ivejp) [reply pkg/gateway/api/metrics_event.go:500]: Not anymore. I have confirmed that all projects are using the latest SDK.

**Review by ono-max — APPROVED** [review]
👍

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**